sentry_sdk/worker.py

```diff
@@ -67,8 +67,14 @@ def start(self):
                     target=self._target, name="raven-sentry.BackgroundWorker"
                 )
                 self._thread.daemon = True
-                self._thread.start()
-                self._thread_for_pid = os.getpid()
+                try:
+                    self._thread.start()
+                    self._thread_for_pid = os.getpid()
+                except RuntimeError:
+                    # At this point we can no longer start because the interpreter
+                    # is already shutting down.  Sadly at this point we can no longer
```

Comment:
Does this mean that e.g. a CLI tool that exits with an exception won't be able to log the error that caused it to terminate to Sentry?

If yes, wouldn't it be possible to launch a background *process* in this case, send the data that needs to be sent to it via stdin, and then let the main process exit while the child process takes care of sending any remaining events.

Reply:
Not necessarily. When we release a fix for this you can test if it works for our CLI tool. If not we will have a look. 
(changing from threads to processes would be a big change in the SDKs architecture and we will not do this)

Reply:
It's not a really strong requirement for me, I was just wondering because I saw "waiting to send events" in some cases when a CLI crashed with an exception